trio/testing/_fake_net.py

```diff
@@ -12,9 +12,19 @@
 import errno
 import ipaddress
 import os
-from typing import TYPE_CHECKING, Optional, Union
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Iterable,
+    NoReturn,
+    Optional,
+    TypeVar,
+    Union,
+    overload,
+)
 
 import attr
+from typing_extensions import Buffer, Self
```

Comment:
We probably want to guard this import, since `trio.testing` is public.

Reply:
hmm, this should ideally be picked up by the `no test-requirements` CI run. Will look at it.

Reply:
and indeed it did :tada: 
https://github.com/python-trio/trio/actions/runs/6567804909/job/17841165556?pr=2820#step:5:720
I just didn't see it because all runs were failing